src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -323,7 +325,16 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
                 return false;
             }
 
-            analyzerConfigSet = AnalyzerConfigSet.Create(configs);
+            analyzerConfigSet = AnalyzerConfigSet.Create(configs, out var unsetGlobalKeys);
+            foreach (var unsetKey in unsetGlobalKeys)
+            {
+                diagnostics.Add(Diagnostic.Create(
```

Comment:
I think we could let the AnalyzerConfigSet produce the diagnostics and then expose them through a Diagnostics property. We could then add those diagnostics to the bag here.

Reply:
Yeah, I actually implemented it that way initially. It just felt kind of strange as if you're were hosting it yourself (it's a public type) you couldn't actually get at the information about the keys that were unset, just somewhat opaque diagnostics.

---
In reply to: [421831468](https://github.com/dotnet/roslyn/pull/43889#discussion_r421831468) [](ancestors = 421831468)

Reply:
Ok, the final commit goes the way of not returning the list, but instead just taking the diagnostic bag in. It definitely simplifies things. I've made it internal for now, we can always make a public property that exposes them if there is demand. 

---
In reply to: [421847650](https://github.com/dotnet/roslyn/pull/43889#discussion_r421847650) [](ancestors = 421847650,421831468)